scripts/docs/json_builder.py

```diff
@@ -0,0 +1,344 @@
+import importlib
+import inspect
+import json
+import os
+import pkgutil
+import re
+import sys
+from types import FunctionType, ModuleType
+from typing import Any, Dict, Optional
+
+import cattrs
+import docstring_parser
+from attr import define
+from docstring_parser import parse
+from docstring_parser.common import DocstringStyle
+
+
+# Object definitions
+@define
+class ParamsData:
+    arg_name: str
+    default: str | None
+    is_optional: bool | None
+    type_name: str | None
+    description: str | None
+
+
+@define
+class ReturnData:
+    type_name: str | None
+    description: str | None
+    return_name: str | None
+
+
+@define
+class DocstringData:
+    params: list[ParamsData]
+    long_description: str | None
+    short_description: str | None
+    examples: str
+    returns: Optional[ReturnData] = None
```

Comment:
`Optional[ReturnData]` is equivalent to `ReturnData | None`. Please use only one approach and do not mix both of them.